apps/ws/src/socket/usecases/external-services-route/external-services-route.usecase.ts

```diff
@@ -12,42 +11,41 @@ const LOG_CONTEXT = 'ExternalServicesRoute';
 
 @Injectable()
 export class ExternalServicesRoute {
-  public readonly bullMqService: BullMqService;
+  constructor(private wsGateway: WSGateway, private messageRepository: MessageRepository) {}
 
-  constructor(private wsGateway: WSGateway, private messageRepository: MessageRepository) {
-    this.bullMqService = new BullMqService();
-  }
   public async execute(command: ExternalServicesRouteCommand) {
-    if (command.event === WebSocketEventEnum.UNSEEN) {
-      await this.sendUnseenCountChange(command);
-
-      return;
-    }
-
-    if (command.event === WebSocketEventEnum.UNREAD) {
-      await this.sendUnreadCountChange(command);
-
-      return;
-    }
+    const isOnline = await this.connectionExist(command);
+    if (isOnline) {
+      if (command.event === WebSocketEventEnum.RECEIVED) {
+        await this.processReceivedEvent(command);
+      }
 
-    if (command.event === WebSocketEventEnum.RECEIVED) {
-      // TODO: Retro-compatibility for a bit just in case stalled messages
-      if (command.payload?.message) {
-        Logger.verbose('Sending full message in the payload', LOG_CONTEXT);
-        await this.wsGateway.sendMessage(command.userId, command.event, command.payload);
+      if (command.event === WebSocketEventEnum.UNSEEN) {
+        await this.sendUnseenCountChange(command);
+      }
 
-        return;
+      if (command.event === WebSocketEventEnum.UNREAD) {
+        await this.sendUnreadCountChange(command);
```

Comment:
Headless sends this events to request the count changes, so keeping them.

Reply:
The `headless` and `notification-center` actually are listening for these events. The events are triggered when the messages are marked as read/seen through the API or the `notification-center`.